packages/gatsby-source-contentful/src/rich-text.js

```diff
@@ -1,184 +1,69 @@
-const _ = require(`lodash`)
-const { BLOCKS, INLINES } = require(`@contentful/rich-text-types`)
-
-const isEntryReferenceNode = node =>
-  [
-    BLOCKS.EMBEDDED_ENTRY,
-    INLINES.ENTRY_HYPERLINK,
-    INLINES.EMBEDDED_ENTRY,
-  ].indexOf(node.nodeType) >= 0
-
-const isAssetReferenceNode = node =>
-  [BLOCKS.EMBEDDED_ASSET, INLINES.ASSET_HYPERLINK].indexOf(node.nodeType) >= 0
-
-const isEntryReferenceField = field => _.get(field, `sys.type`) === `Entry`
-const isAssetReferenceField = field => _.get(field, `sys.type`) === `Asset`
-
-const getFieldProps = (contentType, fieldName) =>
-  contentType.fields.find(({ id }) => id === fieldName)
-
-const getAssetWithFieldLocalesResolved = ({ asset, getField }) => {
-  return {
-    ...asset,
-    fields: _.mapValues(asset.fields, getField),
-  }
-}
-
-const getFieldWithLocaleResolved = ({
-  field,
-  contentTypesById,
-  getField,
-  defaultLocale,
-  resolvedEntryIDs,
-}) => {
-  // If the field is itself a reference to another entry, recursively resolve
-  // that entry's field locales too.
-  if (isEntryReferenceField(field)) {
-    if (resolvedEntryIDs.has(field.sys.id)) {
-      return field
+import { documentToReactComponents } from "@contentful/rich-text-react-renderer"
+import resolveResponse from "contentful-resolve-response"
+
+// We need the original id from contentful as id attribute
+// This code can be simplified as soon we properly replicate
+// Contentful's sys object
+const prepareIdsForResolving = obj => {
+  for (let k in obj) {
+    const value = obj[k]
+    if (
```

Comment:
Nit: Similar as commented earlier, this `if` could be inside the `typeof` check of the else branch.